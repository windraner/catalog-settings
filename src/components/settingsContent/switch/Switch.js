import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './switch.css';

export default class Switch extends Component {
  changeHandler = () => {
    const { name } = this.props;
    this.props.handlerCallBack(name);
  }

  render() {
    const { isChecked } = this.props;

    return (
      <label className="catalog-settings-content__switch-label">
        <input
          checked={isChecked}
          type="checkbox"
          className="catalog-settings-content__switch-input"
          onChange={this.changeHandler}
        />
        <span className="catalog-settings-content__switch-text">{isChecked ? 'On' : 'Off'}</span>
      </label>
    );
  }
}

Switch.propTypes = {
  isChecked: PropTypes.bool.isRequired,
  handlerCallBack: PropTypes.func.isRequired
};
