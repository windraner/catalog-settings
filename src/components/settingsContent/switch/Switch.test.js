import React from 'react';
import Switch from './Switch';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<Switch />', () => {
  it('Check switcher input', () => {
    const wrapper = shallow(<Switch isChecked={true} handlerCallBack={function() {}} />);
    expect(wrapper.find('input').props().checked).toBe(true);
    wrapper.setProps({isChecked: false})
    expect(wrapper.find('input').props().checked).toBe(false);
  });
  it('Check switcher text', () => {
    const wrapper = shallow(<Switch isChecked={false} handlerCallBack={function() {}} />);
    expect(wrapper.find('.catalog-settings-content__switch-text').text()).toBe('Off');
    wrapper.setProps({isChecked: true})
    expect(wrapper.find('.catalog-settings-content__switch-text').text()).toBe('On');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<Switch isChecked={true} handlerCallBack={function() {}} />)
    expect(toJson(tree)).toMatchSnapshot()
  })
});
