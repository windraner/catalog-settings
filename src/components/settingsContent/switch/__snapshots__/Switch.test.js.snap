// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> matches the snapshot 1`] = `
<label
  className="catalog-settings-content__switch-label"
>
  <input
    checked={true}
    className="catalog-settings-content__switch-input"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="catalog-settings-content__switch-text"
  >
    On
  </span>
</label>
`;
